hummingbot/data_feed/market_data_provider.py

```diff
@@ -105,49 +101,83 @@ async def update_rates_task(self):
                     break
 
                 rate_oracle = RateOracle.get_instance()
+
+                # Separate gateway and non-gateway connectors
+                gateway_tasks = []
+                gateway_task_metadata = []  # Store (connector_pair, trading_pair) for each task
+                non_gateway_connectors = {}
+
                 for connector, connector_pairs in self._rates_required.items():
-                    if connector == "gateway":
-                        tasks = []
+                    # Detect gateway connectors: either new format (contains "/") or old format (contains "gateway")
```

Comment:
I think this is fine for now, but we should clean this up when we do the refactor.